hpx/runtime/serialization/array.hpp

```diff
@@ -99,6 +98,13 @@ namespace hpx { namespace serialization
         return array<T>(begin, size);
     }
 
+    // implement serialization for arrays
+    template <class Archive, class T, std::size_t N>
+    void serialize(Archive& ar, T (&a)[N], const unsigned int /* version */)
+    {
+        ar & hpx::serialization::make_array(a, N);
+    }
+
     // implement serialization for boost::array
```

Comment:
I'd prefer if you decided to always serialize arrays in your code by using `make_array()` explicitly instead of the changes proposed. 

Here is my (not very strong) rationale: we currently don't allow things like this:

    char buffer[N];
    hpx::async<Action>(id, buffer);

as this could imply that the user wants to 'return' data from the action through the argument. This is not supported by HPX. Allowing direct serialization of arrays using HPX functionality somehow 'feels' to relate to that.